File: inception/class_script.php

<?php

$a = 7;

class Human
{
    
    public string $name;
    public int $age;
    public string $sex; 
    
    public function display()
    {
        echo "\nName: $this->name"; 
        echo "\nAge: $this->age";
        echo "\nSex: $this->sex";
    }
}

class Hardware
{
    public int $year;
    public string $type;
    public string $material; 
    
    public function show()
    {
        echo "\n\nType: $this->type";
        echo "\nYear: $this->year";
        echo "\nMaterial: $this->material";
    }
}

$person = new Human();

$person->name = "Yeji";
$person->age = 26;
$person->sex = "Female";

$com = new Hardware();

$com->type = "Desktop";
$com->year = 1978;
$com->material = "Silicon";

if($a > 10)
{
    $person->display();
}

else
{
    $com->show();
}


?>

File: inception/class_script.pl
#!/usr/bin/perl
use strict;
use warnings;

my $a = 7;

# Define the Human class
package Human {
    use base 'Exporter';
    our @EXPORT_OK = qw/new/;

    sub new {
        my ($class, %args) = @_;
        my $self = {
            name => $args{name} // '',
            age => $args{age}  // 0,
            sex => $args{sex}  // '',
        };
        bless $self, $class;
        return $self;
    }

    sub display {
        my $self = shift;
        print "\nName: $self->{name}";
        print "\nAge: $self->{age}";
        print "\nSex: $self->{sex}";
        print "\n";
    }
}

# Define the Hardware class
package Hardware {
    sub new {
        my ($class, %args) = @_;
        my $self = {
            year => $args{year}     // 0,
            type => $args{type}     // '',
            material => $args{material} // '',
        };
        bless $self, $class;
        return $self;
    }

    sub show {
        my $self = shift;
        print "\n\nType: $self->{type}";
        print "\nYear: $self->{year}";
        print "\nMaterial: $self->{material}";
        print "\n";
    }
}

# Main script execution
package main;

# Create instances
my $person = Human->new(name => "Yeji", age => 26, sex => "Female");
my $com = Hardware->new(type => "Desktop", year => 1978, material => "Silicon");

if ($a > 10) {
    $person->display();
} 
else {
    $com->show();
}
